sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -87,7 +87,7 @@ trait NamedExpression extends Expression {
 
   protected def typeSuffix =
     if (resolved) {
-      dataType match {
+      prettyDataType match {
```

Comment:
This line appears to be the only code that consumes the new `prettyDataType` attribute